### DISCUSSION TITLE
Trying to package for Nixpkgs

### DISCUSSION THREAD

**Discussion by Luflosi** [pradyunsg/furo#197] [Potential Issue] [answered]
Hello,
I'm trying to package furo in [Nixpkgs](https://github.com/NixOS/nixpkgs) so that I can build [kitty](https://github.com/kovidgoyal/kitty) with Nix, since kitty now uses furo for its documentation. I downloaded the latest version 2021.07.05.beta38 from GitHub. The current versions of the dependencies in Nixpkgs are Sphinx 4.0.2, beautifulsoup 4.9.3 and Python 3.8.9.
The packaging process itself works. Unfortunately, the result seems to be broken somehow. When I then try to import the furo module, I get the following error message:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/nix/store/mgizfi86wdix89hgc5g4kw14vdhvd8sv-python3.8-furo-2021.07.05.beta38/lib/python3.8/site-packages/furo/__init__.py", line 119, in <module>
    app: sphinx.application.Sphinx,
AttributeError: module 'sphinx' has no attribute 'application'
```
Do you know why this could be happening? I'm hoping I made some stupid and obvious mistake. I'm happy to provide any additional information that may be helpful.
<details open>
<summary>Here is the Nix code I wrote so far, in case it helps.</summary>

```nix
{ lib
, buildPythonPackage
, pythonOlder
, fetchFromGitHub
, sphinx
, beautifulsoup4
}:

buildPythonPackage rec {
  pname = "furo";
  version = "2021.07.05.beta38";
  format = "flit";
  disable = pythonOlder "3.6";

  src = fetchFromGitHub {
    owner = "pradyunsg";
    repo = "furo";
    rev = version;
    sha256 = "sha256-dyhxn/l5X4g6qYvXpR1riCbhAPVweXWRaDoxXAo+KFg=";
  };

  propagatedBuildInputs = [
    sphinx
    beautifulsoup4
  ];

  pythonImportsCheck = [ "furo" ];

  meta = with lib; {
    description = "A clean customizable documentation theme for Sphinx";
    homepage = "https://github.com/pradyunsg/furo";
    license = licenses.mit;
    maintainers = with maintainers; [ Luflosi ];
  };
}
```
</details>

**Comment by pradyunsg** [comment, marked as answer]
You seem to have not declared that Furo depends on Sphinx at runtime?

**Reply by Luflosi** [reply]
I did. That same `__init__.py` file even has the statement `import sphinx` pretty much at the top, before the line that fails, and this succeeds, at least there is no error when importing Sphinx. Also, the error message does not complain that Sphinx could not be found but that the "module 'sphinx' has no attribute 'application'". If I replace all occurrences of "sphinx.application.Sphinx" with nothing, importing furo works fine but this might cause other problems, does it?

**Reply by Luflosi** [reply]
I managed to package furo now. I updated to the latest version and am now using the sources from PyPI. I did not realise, that I can't just use the code GitHub directly because of the required build step.
Thank you for your time.
